pinot-controller/src/main/java/org/apache/pinot/controller/recommender/rules/RulesToExecute.java

```diff
@@ -0,0 +1,140 @@
+        case VariedLengthDictionaryRule:
+          return new VariedLengthDictionaryRule(inputManager, outputManager);
+        default:
+          return null;
+      }
+    }
+  }
+  // All rules will execute by default unless explicitly specifying "recommendInvertedSortedIndexJoint" = "false"
+  boolean _recommendKafkaPartition = DEFAULT_RECOMMEND_KAFKA_PARTITION;
+  boolean _recommendPinotTablePartition = DEFAULT_RECOMMEND_PINOT_TABLE_PARTITION;
+  boolean _recommendInvertedSortedIndexJoint = DEFAULT_RECOMMEND_INVERTED_SORTED_INDEX_JOINT;
+  boolean _recommendBloomFilter = DEFAULT_RECOMMEND_BLOOM_FILTER;
+  boolean _recommendNoDictionaryOnHeapDictionaryJoint = DEFAULT_RECOMMEND_NO_DICTIONARY_ONHEAP_DICTIONARY_JOINT;
+  boolean _recommendVariedLengthDictionary = DEFAULT_RECOMMEND_VARIED_LENGTH_DICTIONARY;
+  boolean _recommendFlagQuery = DEFAULT_RECOMMEND_FLAG_QUERY;
+
+  @JsonSetter(nulls = Nulls.SKIP)
+  public void setRecommendVariedLengthDictionary(boolean recommendVariedLengthDictionary) {
+    _recommendVariedLengthDictionary = recommendVariedLengthDictionary;
+  }
+
+  @JsonSetter(nulls = Nulls.SKIP)
+  public void setRecommendFlagQuery(boolean recommendFlagQuery) {
+    _recommendFlagQuery = recommendFlagQuery;
+  }
+
+  @JsonSetter(nulls = Nulls.SKIP)
+  public void setRecommendNoDictionaryOnHeapDictionaryJoint(boolean recommendNoDictionaryOnHeapDictionaryJoint) {
+    _recommendNoDictionaryOnHeapDictionaryJoint = recommendNoDictionaryOnHeapDictionaryJoint;
+  }
+
+  @JsonSetter(nulls = Nulls.SKIP)
+  public void setRecommendKafkaPartition(boolean recommendKafkaPartition) {
+    _recommendKafkaPartition = recommendKafkaPartition;
+  }
+
+  @JsonSetter(nulls = Nulls.SKIP)
+  public void setRecommendInvertedSortedIndexJoint(boolean recommendInvertedSortedIndexJoint) {
+    _recommendInvertedSortedIndexJoint = recommendInvertedSortedIndexJoint;
+  }
+
+  @JsonSetter(nulls = Nulls.SKIP)
+  public void setRecommendPinotTablePartition(boolean recommendPinotTablePartition) {
+    _recommendPinotTablePartition = recommendPinotTablePartition;
+  }
+
+  @JsonSetter(nulls = Nulls.SKIP)
+  public void setRecommendBloomFilter(boolean recommendBloomFilter) {
+    _recommendBloomFilter = recommendBloomFilter;
+  }
+
+  public boolean isRecommendVariedLengthDictionary() {
+    return _recommendVariedLengthDictionary;
+  }
+
+  public boolean isRecommendFlagQuery() {
+    return _recommendFlagQuery;
+  }
+
+  public boolean isRecommendNoDictionaryOnHeapDictionaryJoint() {
+    return _recommendNoDictionaryOnHeapDictionaryJoint;
+  }
+
+  public boolean isRecommendKafkaPartition() {
+    return _recommendKafkaPartition;
+  }
+
+  public boolean isRecommendInvertedSortedIndexJoint() {
+    return _recommendInvertedSortedIndexJoint;
+  }
+
+  public boolean isRecommendPinotTablePartition() {
+    return _recommendPinotTablePartition;
+  }
+
+  public boolean isRecommendBloomFilter() {
+    return _recommendBloomFilter;
+  }
+
+  public enum Rules {
```

Comment:
The enum name shouldn't really have "plural". The enum although defines multiple constants, it represents only 1 of them at a give time. So, we should simply call it Rule?

Reply:
done